File: SmartPhone++/SmartPhone.Services/Database/ProductImage.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace SmartPhone.Services.Database
{
    public class ProductImage
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        public byte[] ImageData { get; set; } = new byte[0];
        
        [MaxLength(50)]
        public string? FileName { get; set; }
        
        [MaxLength(20)]
        public string? ContentType { get; set; }
        
        [MaxLength(200)]
        public string? AltText { get; set; }
        
        public bool IsPrimary { get; set; } = false;
        
        public int DisplayOrder { get; set; } = 0;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        // Foreign key
        public int ProductId { get; set; }
        
        // Navigation property
        public Product Product { get; set; } = null!;
    }
} 

File: SmartPhone++/UI/smartphone_mobile_client/lib/model/product_image.dart
import 'dart:convert';
import 'package:json_annotation/json_annotation.dart';

part 'product_image.g.dart';

@JsonSerializable()
class ProductImage {
  final int id;
  final String? imageData; // Base64 string from backend
  final String? fileName;
  final String? contentType;
  final String? altText;
  final bool isPrimary;
  final int displayOrder;
  final DateTime createdAt;
  final int productId;

  ProductImage({
    this.id = 0,
    this.imageData,
    this.fileName,
    this.contentType,
    this.altText,
    this.isPrimary = false,
    this.displayOrder = 0,
    required this.createdAt,
    this.productId = 0,
  });

  factory ProductImage.fromJson(Map<String, dynamic> json) => _$ProductImageFromJson(json);
  Map<String, dynamic> toJson() => _$ProductImageToJson(this);
} 